test_runner/fixtures/utils.py

```diff
@@ -146,9 +147,13 @@ def global_counter() -> int:
     This is useful for giving output files a unique number, so if we run the
     same command multiple times we can keep their output separate.
     """
-    global _global_counter
-    _global_counter += 1
-    return _global_counter
+    global _global_counter, _global_counter_lock
+    _global_counter_lock.acquire()
+    try:
+        _global_counter += 1
+        return _global_counter
+    finally:
+        _global_counter_lock.release()
```

Comment:
```suggestion
    global _global_counter, _global_counter_lock
    with _global_counter_lock:
        _global_counter += 1
        return _global_counter
```

Reply:
Feels like there ought to be a function call, but according to https://docs.python.org/3/library/threading.html#with-locks nope.